### PR TITLE
Support PHP8 and update dependencies

### LINKED ISSUE #3: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by TavoNiievez**
Closes #3.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,10 @@
+/.idea/
```
Comment (Ocramius) [inline .gitignore]: This should not be part of the library: please use a global `.gitignore` as described in https://docs.github.com/en/github/using-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer

File: phpcs.xml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0"?>
 <ruleset name="Laminas coding standard">
-    <rule ref="./vendor/laminas/laminas-coding-standard/ruleset.xml"/>
+    <rule ref="./vendor/laminas/laminas-coding-standard/src/LaminasCodingStandard/ruleset.xml"/>
```
Comment (Ocramius) [inline phpcs.xml]: Should probably be `<rule ref="LaminasCodingStandard">`

File: phpunit.xml.dist [hunk 3]
```diff
@@ -1,17 +1,13 @@
...
-            <directory>./test/</directory>
-        </testsuite>
-    </testsuites>
-
-    <filter>
-        <whitelist processUncoveredFilesFromWhitelist="true">
-            <directory suffix=".php">src</directory>
-        </whitelist>
-    </filter>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd" bootstrap="./vendor/autoload.php" colors="true">
```
Comment (Ocramius) [inline phpunit.xml.dist]: The XSD should point to `vendor/phpunit/phpunit/phpunit.xsd`, and not some HTTP address

File: test/Xml2JsonTest.php [hunk 4]
```diff
@@ -454,33 +455,34 @@ public function testXmlToJsonWithXmlContainingSelfClosingElement()
...
     {
+        $this->expectException(RecursionException::class);
         Xml2Json::$maxRecursionDepthAllowed = 1;
         Xml2Json::fromXml($xmlStringContents, true);
     }
 
     /**
      * @group Laminas-11385
      * @dataProvider providerNestingDepthIsHandledProperly
+     * @param $xmlStringContents
```
Comment (Ocramius) [inline test/Xml2JsonTest.php]: An `@param` is kinda useless if no information comes with it: can we specify the type here? perhaps adding `string` to the native type, and dropping `@param` overall?

File: test/Xml2JsonTest.php [hunk 5]
```diff
@@ -454,33 +455,34 @@ public function testXmlToJsonWithXmlContainingSelfClosingElement()
 
     /**
      * @group Laminas-11385
-     * @expectedException Laminas\Xml2Json\Exception\RecursionException
      * @dataProvider providerNestingDepthIsHandledProperly
+     * @param $xmlStringContents
```
Comment (Ocramius) [inline test/Xml2JsonTest.php]: An `@param` is kinda useless if no information comes with it: can we specify the type here? perhaps adding `string` to the native type, and dropping `@param` overall?

File: src/Xml2Json.php [hunk 6]
```diff
@@ -109,10 +115,10 @@ protected static function getXmlValue($simpleXmlElementObject)
      * @param bool $ignoreXmlAttributes
      * @param int $recursionDepth
      * @return array
-     * @throws Exception\RecursionException if the XML tree is deeper than the
+     * @Throws Exception\RecursionException if the XML tree is deeper than the
```
Comment (Ocramius) [inline src/Xml2Json.php]: `@throws` (lowercase) was correct here

**Review by Ocramius — CHANGES_REQUESTED** [review]
Overall, comments are mostly about minor details, but easily adjusted :-)

**Comment by TavoNiievez** [comment]
@Ocramius Sorry, I got carried away by `phpcs` and `PHPStorm`. Made all requested changes, thanks for the review 😃

**Review by Ocramius — APPROVED** [review]
LGTM 👍

**Comment by Ocramius** [comment]
Thanks @TavoNiievez!

**Pull request merged**